src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java

```diff
@@ -410,6 +384,50 @@ private JButton getJButtonClose() {
 		return jButtonClose;
 	}
 
+	private JButton getJButtonDelete() {
+		if (jButtonDelete == null) {
+			jButtonDelete = new JButton(MessageBundle.getMessage("angal.common.delete.btn"));
+			jButtonDelete.setMnemonic(MessageBundle.getMnemonic("angal.common.delete.btn.key"));
+			jButtonDelete.setVisible(false);
+			jButtonDelete.addActionListener(actionEvent -> {
+				
+				if (jTableOutcomes.getSelectedRowCount() > 1 ) {
+					MessageDialog.error(this, "angal.medicalstock.pleaseselectonlyonemovement.msg");
+					return;
+				}
+				int selectedRow = jTableOutcomes.getSelectedRow();
+				if (selectedRow == -1) {
+					MessageDialog.error(this, "angal.medicalstock.pleaseselectoucomemovement.msg");
+					return;
+				}
+				MovementWard selectedMovement = (MovementWard)jTableOutcomes.getValueAt(selectedRow, -1);
+				try {
+					MovementWard movWard = movWardBrowserManager.getLastMovementWard(wardSelected);
+					if (movWard.getCode() == selectedMovement.getCode()) {
+						int delete = MessageDialog.yesNo(null, "angal.medicalstock.doyoureallywanttodeletethismovement.msg");
+						if (delete == JOptionPane.YES_OPTION) {
+							movWardBrowserManager.deleteLastMovementWard(movWard);
```

Comment:
Does this need to be in try-catch block in case an exception is thrown allowing for outputting a failure message?

Reply:
it is

Reply:
or do you mean another thing?

Reply:
I missed the outer try-catch; sorry.   I typically try to put them as narrow as possible but this is fine.   You can mark it resolved.